pkg/common/helper.go

```diff
@@ -585,3 +585,15 @@ func PopulateFieldsFromValues[T string | bool | int](fieldsToValues map[*T]T) {
 		}
 	}
 }
+
+// PopulateMapWithMap iterates over the key-value pairs in the populateMap and adds them
+// to the mainMap only if the key does not already exist in the mainMap. If a key
+// already exists, the value from the populateMap is not added
+func PopulateMapWithMap(mainMap, populateMap map[string]string) map[string]string {
+	for key, value := range populateMap {
+		if _, ok := mainMap[key]; !ok {
+			mainMap[key] = value
+		}
+	}
+	return mainMap
+}
```

Comment:
can be simplified by using `labels.Merge(populateMap, mainMap)`

Reply:
@liranbg yes, but it will be less effective because lets say we have:

```
function's selector - n elements
project's selector - k elements
platform selector - m elements
```

for my solution complexity will be O(k) + O(n) (if we think that check if key is in the map is O(1))
for labels.Merge it will be O(n)+O(k) on 1st merge and O(n+k)(worst case)+O(m) on the 2nd merge

Reply:
well, we're talking labels here, there wont be more than 100 keys at the worst scenario
good thinking on the complexity aspect, but irrelevant for such a small amount of items